Dockerfile

```diff
@@ -2,21 +2,49 @@ FROM centos
 
 RUN echo "multilib_policy=best" >> /etc/yum.conf
 RUN yum  -y update && \
-	yum install -y gcc-c++ git xinetd perl curl python openssh-server openssh-clients expect man python-argparse sshpass wget make cmake dos2unix which unzip lsof net-tools|| true && \
-	yum install -y http://libslack.org/daemon/download/daemon-0.6.4-1.i686.rpm > /dev/null && \
-	package-cleanup --cleandupes && \
-	yum  -y clean all
+    yum install -y gcc-c++ git xinetd perl curl python openssh-server openssh-clients expect man python-argparse sshpass wget make cmake dos2unix which unzip lsof net-tools || true && \
+    yum install -y http://libslack.org/daemon/download/daemon-0.6.4-1.i686.rpm > /dev/null && \
+    package-cleanup --cleandupes && \
+    yum  -y clean all
 
 RUN ssh-keygen -f /etc/ssh/ssh_host_rsa_key -N '' -t rsa && \
     ssh-keygen -f /etc/ssh/ssh_host_dsa_key -N '' -t dsa && \
     ssh-keygen -t ecdsa -N "" -f /etc/ssh/ssh_host_ecdsa_key && \
     ssh-keygen -t ed25519 -N "" -f /etc/ssh/ssh_host_ed25519_key && \
     sed -i 's/PermitRootLogin without-password/PermitRootLogin yes/' /etc/ssh/sshd_config && \
-	sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd && \
-	echo 'root:docker' | chpasswd
+    sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd && \
+    echo 'root:docker' | chpasswd
 
 WORKDIR /opt/vista
-ADD . /opt/vista/
+# Add each folder individually to improve rebuild times
+ADD ./Cache /opt/vista/Cache
+ADD ./cache-files /opt/vista/cache-files
+ADD ./Common /opt/vista/Common
+ADD ./Dashboard /opt/vista/Dashboard
+ADD ./EWD /opt/vista/EWD
+ADD ./GTM /opt/vista/GTM
+ADD ./tests /opt/vista/tests
+ADD ./test.cmake /opt/vista/
+ADD ./ViViaN /opt/vista/ViViaN
+ADD ./*.sh /opt/vista/
+
+ARG instance=osehra
+ARG postInstallScript="-p ./Common/ovydbPostInstall.sh"
+ARG flags="-y -b -e -m"
+ENV instance_name=$instance
+ENV install_flags="$flags -i ${instance_name} ${postInstallScript}"
+
+RUN dos2unix /opt/vista/* && \
+    dos2unix /opt/vista/Cache/* && \
+    dos2unix /opt/vista/Cache/etc/init.d/* && \
+    dos2unix /opt/vista/Common/* && \
+    dos2unix /opt/vista/Dashboard/* && \
+    dos2unix /opt/vista/EWD/* && \
+    dos2unix /opt/vista/EWD/etc/init.d/* && \
+    dos2unix /opt/vista/GTM/* && \
+    dos2unix /opt/vista/GTM/bin/* && \
+    dos2unix /opt/vista/GTM/etc/init.d/* && \
+    dos2unix /opt/vista/ViViaN/*
```

Comment:
Is this needed for all versions of VistA, or can the dos2unix command inside `ViViaN/vivianInstall.sh` be updated from ./VistA-M to /opt/vista?

Reply:
It is required if you run docker on windows. I'm happy this is back. If it is ran on a unix system it effectively becomes a noop